cmd/loogtui/main.go

```diff
@@ -190,3 +167,48 @@ func runCollector(
 		}
 	}
 }
+
+func loadHistoryFromDB(rps store.ResourcePatchStore, trackerService *service.TrackerService, p *tea.Program) error {
+	objectRevisionState := map[string]*store.Snapshot{}
+	err := rps.WalkObjectRevisions(func(
+		objectUID string,
+		revisionID store.RevisionID,
+		snapshot *store.Snapshot,
+		patch *store.Patch,
+	) bool {
+		var current *store.Snapshot
+		if snapshot != nil {
+			// full snapshot: start anew
+			diffMap := make(diffmap.DiffMap)
+			maps.Copy(diffMap, snapshot.Object)
+			current = &store.Snapshot{
+				ID:     revisionID,
+				Object: diffMap,
+				Time:   snapshot.Time,
+			}
+		} else {
+			// patch: apply on top of last state
+			base := make(diffmap.DiffMap)
+			maps.Copy(base, objectRevisionState[objectUID].Object)
+			diffmap.Apply(base, patch.Patch)
+			current = &store.Snapshot{
+				ID:     revisionID,
+				Object: base,
+				Time:   patch.Time,
+			}
+		}
+		objectRevisionState[objectUID] = current
+		trackerService.WarmCache(objectUID, current)
+		p.Send(ui.NewCommitCommand(
+			objectUID,
+			current.Object["kind"].(string),
```

Comment:
use `unstructured` getters, e.g.:

```go
u := &unstructured.Unstructured{Object: current.Object}
u.GetKind()

Reply:
fixed in a94a949a661cb42b2b557619970ed5aee71a6c9d